src/core/dbus-execute.c

```diff
@@ -225,6 +225,45 @@ static int property_get_cpu_affinity(
         return sd_bus_message_append_array(reply, 'y', array, allocated);
 }
 
+static int property_get_numa_mask(
+                sd_bus *bus,
+                const char *path,
+                const char *interface,
+                const char *property,
+                sd_bus_message *reply,
+                void *userdata,
+                sd_bus_error *error) {
+
+        ExecContext *c = userdata;
+        _cleanup_free_ uint8_t *array = NULL;
+        size_t allocated;
+
+        assert(bus);
+        assert(reply);
+        assert(c);
+
+        (void) cpu_set_to_dbus(&c->numa_policy.nodes, &array, &allocated);
+
+        return sd_bus_message_append_array(reply, 'y', array, allocated);
+}
+
+static int property_get_numa_policy(
+                sd_bus *bus,
+                const char *path,
+                const char *interface,
+                const char *property,
+                sd_bus_message *reply,
+                void *userdata,
+                sd_bus_error *error) {
+        ExecContext *c = userdata;
+
+        assert(bus);
+        assert(reply);
+        assert(c);
+
+        return sd_bus_message_append_basic(reply, 'i', &c->numa_policy.type);
```

Comment:
I think it'd be better to pass the policy over dbus as a string ("default", "bind", ...). It would be much more user-friendly and would make introspection and use from scripts easier to understand.

Reply:
I discussed this with @poettering and his guidance was to pass everything over dbus in parsed form, i.e. no stringy APIs. I don't care either way, but you guys should agree on general guidance.

Reply:
@poettering?

Reply:
hmm, difficult question. Generally, when we have enums with number assignments defined by others that are API, then we should just use the numbers (example: waitid() CLD_XXX constants, or unix SIGXXX signal constants). If we have internal stuff however, or any other stuff where the numbers really shouldn't be exposed, then we shouldn't expose them, and use strings instead. 

And then there are things in between: sometimes we want to use official stuff, but with special "invalid"  or "default" values. in that case strings might be better, as then we could use the empty string for that.

So dunno, it really depends on what this enum is like here. 

If in doubt we should probably opt for strings rather than integers...